components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceOrganizationServiceImpl.java

```diff
@@ -0,0 +1,250 @@
+                String msg = "No children connected to device";
+                log.error(msg);
+                return Response.status(Response.Status.NOT_FOUND).entity(msg).build();
+            }
+            return Response.status(Response.Status.OK).entity(result).build();
+        } catch (DeviceOrganizationException e) {
+            String msg = "Error occurred while retrieving children of device";
+            log.error(msg, e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(msg).build();
+        } catch (InvalidConfigurationException e) {
+            log.error("failed to add operation", e);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
+        }
+    }
+
+    @GET
+    @Path("/nodes")
+    @Override
+    public Response generateNodes() {
+        List<DeviceOrganizationVisNode> result;
+        DeviceOrganizationProviderService dops = new DeviceOrganizationProviderServiceImpl();
+        result = dops.generateNodes();
+        if (result.isEmpty()) {
+            String msg = "No Devices enrolled and visible";
+            log.error(msg);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(msg).build();
+        }
+        return Response.status(Response.Status.OK).entity(result).build();
+    }
+
+    @GET
+    @Path("/edges")
+    @Override
+    public Response generateEdges() {
+        List<DeviceOrganizationVisEdge> result;
+        DeviceOrganizationProviderService dops = new DeviceOrganizationProviderServiceImpl();
+        result = dops.generateEdges();
+        if (result.isEmpty()) {
+            String msg = "No Edge connections available and visible";
+            log.error(msg);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(msg).build();
+        }
+        return Response.status(Response.Status.OK).entity(result).build();
+    }
+
+    @GET
+    @Path("/hierarchy")
+    @Override
+    public Response generateHierarchy() {
+        return null;
+    }
+
+    @PUT
+    @Path("/update/{deviceId}/{parentId}")
+    @Override
+    public Response updateDeviceOrganizationParent(@PathParam("deviceId") String deviceId,
+                                                   @PathParam("parentId") String parentId) {
+        if (deviceId == null || parentId == null) {
+            String errorMessage = "One or more parameters are empty.";
+            return Response.status(Response.Status.BAD_REQUEST).entity(errorMessage).build();
+        }
+        String updatedParent;
+        DeviceOrganizationProviderService dops = new DeviceOrganizationProviderServiceImpl();
+        try {
+            updatedParent = dops.updateDeviceOrganizationParent(deviceId,parentId);
+            if (updatedParent == null) {
+                String msg = "Parent not updated";
+                log.error(msg);
+                return Response.status(Response.Status.NOT_FOUND).entity(msg).build();
```

Comment:
IMO, this is not an error. We can log it as a debug log.

Reply:
Fixed ce1b79e04ff7cb44e6a64e88ebab322946c55f47

Reply:
Added checks to see if debug logs enabled